presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -86,12 +90,148 @@ public void testForeignTable()
+            throws SQLException
+    {
+        testReadDecimal(databaseDataType, value, prestoDataType);
+        testWriteDecimal(databaseDataType, value);
+        testCreateDecimal(databaseDataType, value, prestoDataType);
+    }
+
+    private void testReadDecimal(String databaseDataType, BigDecimal value, String prestoDataType)
+            throws SQLException
+    {
+        createTableWithColumn("test_read_decimal", "decimal_column", databaseDataType);
+        try {
+            // Write with PostgreSQL
+            execute(format("INSERT INTO tpch.test_read_decimal (decimal_column) VALUES ('%s')", value));
+
+            // Read with Presto
+            assertQuery("SELECT COLUMN_NAME FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = 'tpch' AND TABLE_NAME = 'test_read_decimal'", "VALUES 'key', 'decimal_column'");
+            assertQuery("SELECT typeOf(decimal_column) FROM test_read_decimal", format("VALUES '%s'", prestoDataType));
+
+            assertQuery("SELECT decimal_column FROM test_read_decimal", format("VALUES CAST('%s' AS %s)", value, prestoDataType));
+            // Sole check of value is insufficient, since testing framework converts expected value to actual value' type. Compare exact (cast to VARCHAR) value too.
+            assertQuery("SELECT CAST(decimal_column AS VARCHAR) FROM test_read_decimal", format("VALUES '%s'", value));
+        }
+        finally {
+            execute("DROP TABLE tpch.test_read_decimal");
+        }
+    }
+
+    private void testWriteDecimal(String databaseDataType, BigDecimal value)
+            throws SQLException
+    {
+        createTableWithColumn("test_write_decimal", "decimal_column", databaseDataType);
+        try {
+            // Write with Presto
+            assertUpdate(format("INSERT INTO test_write_decimal (decimal_column) VALUES (decimal '%s')", value), 1);
+
+            // Read with PostgreSQL
+            BigDecimal read = execute(statement -> {
+                try (ResultSet resultSet = statement.executeQuery("SELECT decimal_column FROM tpch.test_write_decimal")) {
+                    assertTrue(resultSet.next());
+                    return resultSet.getBigDecimal("decimal_column");
+                }
+            });
+            assertTrue(value.compareTo(read) == 0);
+        }
+        finally {
+            execute("DROP TABLE tpch.test_write_decimal");
+        }
+    }
+
+    private void testCreateDecimal(String databaseDataType, BigDecimal value, String prestoDataType)
+            throws SQLException
+    {
+        // Create table and fill it with Presto
+        assertUpdate(format("CREATE TABLE test_create_decimal AS SELECT CAST(DECIMAL '%s' AS %s) AS decimal_column", value, prestoDataType), 1);
+        try {
+            // Check created column data type with PostgreSQL
+            String actualDataType = execute(statement -> {
+                try (ResultSet resultSet = statement.executeQuery(
```

Comment:
Really? Cant you do that simpler?

Like here for example? https://stackoverflow.com/questions/2146705/select-datatype-of-the-field-in-postgres

Reply:
This is simplest way I know of that pulls type _and_ precision/scale/size information. Regarding the link, you can upvote best answer there (i.e. mine:)

I am, however, worried that for other database this won't be possible at all (no `format_type` function) and I will have either write awful `switch` in SQL or (rather) pull separate pieces of information from `information_schema`. However, these considerations I would like to leave for https://github.com/prestodb/presto/issues/9290